ManageAxes.js

```diff
@@ -161,3 +171,80 @@ ManageAxes.resetAxes = async function()
 
     await WSM.APISetLocalCoordinateSystem(nHistoryID, defaultLCS);
 }
+
+// moves all geometry in the current editing history to the world origin, then applies the inverse transform to all instances
+ManageAxes.reOrigin = async () => {
+    // get the current editing group instance path and history ID
+    const editingPath = await FormIt.GroupEdit.GetInContextEditingPath();
+    const editingHistoryId = await FormIt.GroupEdit.GetEditingHistoryID();
+
+    // this action won't be valid for the main history
+    if (editingHistoryId === 0) {
+        await FormIt.UI.ShowNotification("This action requires a group to be edited. \nEdit a group and try again.", FormIt.NotificationType.Error, 0)
+        return;
+    }
+
+    // get the bounding box of the editing history and its upper and lower bounds
+    const editingBBox = await WSM.APIGetBoxReadOnly(editingHistoryId);
+
+    // make a translation transform from the geometry to the world origin using the bounding box
+    const vec3d = await WSM.Geom.Vector3d(-(editingBBox.lower.x + editingBBox.upper.x) / 2, -(editingBBox.lower.y + editingBBox.upper.y) / 2, -editingBBox.lower.z);
+    const geomTransf3d = await WSM.Transf3d.MakeTranslationTransform(vec3d);
+    // get the inverse transform for use later
+    const inverseGeomTransf3d = await WSM.Transf3d.Invert(geomTransf3d);
+
+    // check if the transform from the geom to the world origin is null - if so, stop here
+    const isGeomTransf3dNull = await WSM.Vector3d.IsNull(vec3d)
+    if (isGeomTransf3dNull) {
+        // show a message that nothing was changed
+        await FormIt.UI.ShowNotification("The origin is already centered below this geometry.\nNo changes were made.", FormIt.NotificationType.Information, 0);
+
+        // no need to go any further
+        return;
+    }
+
+    // if we got this far, changes will be made to the model, so start a new undo state
+    await FormIt.UndoManagement.BeginState();
+
+    // get all the non-owned objects in this history
+    const objectIds = await WSM.APIGetAllNonOwnedReadOnly(editingHistoryId);
+    // and move them to the world origin
+    await WSM.APITransformObjects(editingHistoryId, objectIds, geomTransf3d);
+
+    // get all the instances of this history
+    const instancesOfHistory = await WSM.APIGetAllAggregateTransf3dsReadOnly(editingHistoryId, 0);
+
+    // for each instance of this history...
+    for (let i = 0; i < instancesOfHistory.paths.length; i++) {
+        // how many levels "deep" is this instance from history 0
+        const historyDepthIndex = instancesOfHistory.paths[i].ids.length - 1;
+        const instanceObjectHistoryId = instancesOfHistory.paths[i].ids[historyDepthIndex];
+
+        // get the instance's transform and the inverse
+        const instanceTransf3d = instancesOfHistory.transforms[i];
```

Comment:
This is wrong for nested groups. Better probably to use what we talked about get all groups owning the history, then get instances from the group, get the transform on the instance and adjust it.
If you want to keep this (which is more expensive as it does work that is not required), you need to get the transform from the particular instance and adjust that. You are adjusting the aggregate transform, which is an error.

Reply:
I've updated this to get the instances from the group and their respective transforms instead of aggregate transf3d.